File: app/src/main/java/org/mesonet/app/baseclasses/RecyclerViewHolder.java
package org.mesonet.app.baseclasses;

import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;





public abstract class RecyclerViewHolder<TData extends Object, TViewDataBinding extends ViewDataBinding> extends RecyclerView.ViewHolder
{
    private ViewDataBinding mBinding;



    public RecyclerViewHolder(ViewDataBinding inBinding)
    {
        super(inBinding.getRoot());
        mBinding = inBinding;
    }



    public abstract void SetData(TData inData);



    public TViewDataBinding GetBinding()
    {
        try {
            return (TViewDataBinding)mBinding;
        }
        catch (ClassCastException e)
        {
            e.printStackTrace();
        }

        return null;
    }
}

File: app/src/main/java/org/mesonet/app/baseclasses/RecyclerViewHolder.kt
package org.mesonet.app.baseclasses

import android.databinding.ViewDataBinding
import android.support.v7.widget.RecyclerView


abstract class RecyclerViewHolder<in TData, out TViewDataBinding : ViewDataBinding>(protected var mBinding: ViewDataBinding) : RecyclerView.ViewHolder(mBinding.root) {


    abstract internal fun SetData(inData: TData)


    internal fun GetBinding(): TViewDataBinding? {
        try {
            return mBinding as TViewDataBinding
        } catch (e: ClassCastException) {
            e.printStackTrace()
        }

        return null
    }
}
